tests/check_framework/test_security.py

```diff
@@ -535,6 +535,18 @@ def test_low_entropy_secret_key(self):
         self.assertEqual(self.func(None), [base.W009])
 
 
+class CheckSecretKeyValidUnicodeTest(SimpleTestCase):
+    @property
+    def func(self):
+        from django.core.checks.security.base import check_secret_key_valid_unicode
+        return check_secret_key_valid_unicode
+
+    @override_settings(SECRET_KEY=(b'\xffabcdefghijklmnopqrstuvwxyz' * 2))
+    def test_invalid_unicode_secret_key(self):
+        self.assertGreater(len(settings.SECRET_KEY), base.SECRET_KEY_MIN_LENGTH)
+        self.assertEqual(self.func(None), [base.W022])
```

Comment:
You could add a test for the positive case, where the warning is not issued as well.